luxjs/google/module.js

```diff
@@ -37,13 +37,39 @@
                 // Create via element tag
                 // <d3-force data-width=300 data-height=200></d3-force>
                 restrict: 'AE',
-                //
-                link: function (scope, element, attrs) {
-                    if(!scope.googlemaps) {
-                        $lux.log.error('Google maps url not available. Cannot load google maps directive');
-                        return;
-                    }
-                    require([scope.googlemaps], function () {
+                scope: true,
+                controller: function() {
+                    var self = this;
+
+                    // Add a marker to the map
+                    self.addMarker = function(map, marker, location) {
+                        if (marker) {
+                            var gmarker = new google.maps.Marker(angular.extend({
+                                position: location,
+                                map: map,
+                            }, marker));
+                            return gmarker;
+                        }
+                    };
+
+                    // Add marker using lat and lng
+                    self.addLocation = function(map, marker, lat, lng) {
```

Comment:
We should have some consistency with how functions are named, so this should be something like addMarkerByLatLng